core/src/main/scala/pickling/PBuilderReader.scala

```diff
@@ -19,6 +19,10 @@ trait Hintable {
 trait PBuilder extends Hintable {
   def beginEntry(picklee: Any): PBuilder
   def putField(name: String, pickler: PBuilder => Unit): PBuilder
+  def putDynamicFieldNames(names: List[String])(implicit namesPickler: SPickler[List[String]]): PBuilder =
```

Comment:
The order is super important.  It needs to be a minimum of something which denotes order.